File: backend/src/main/scala/domains/contest/objects/response/EvaluateContestAccessResult.scala
package domains.contest.objects.response

import domains.contest.objects.{ContestId, ContestSlug, ContestTitle}

/** 内部权限评估输出，供提交、题目详情等跨 domain 流程复用，不直接产生 HTTP 错误。 */
final case class EvaluateContestAccessResult(
  contestId: ContestId,
  contestSlug: ContestSlug,
  contestTitle: ContestTitle,
  contestStarted: Boolean,
  contestEnded: Boolean,
  isRegistered: Boolean,
  containsProblem: Boolean,
  canViewContest: Boolean,
  canViewContestDetail: Boolean,
  canManageContest: Boolean,
  canViewLinkedContestProblem: Boolean,
  canManageLinkedContestProblem: Boolean,
  canSubmitContestProblem: Boolean
)

File: frontend/src/objects/contest/response/EvaluateContestAccessResult.ts
import type { ContestId } from '@/objects/contest/ContestId'
import type { ContestSlug } from '@/objects/contest/ContestSlug'
import type { ContestTitle } from '@/objects/contest/ContestTitle'

/** 内部比赛访问评估结果；镜像后端 EvaluateContestAccess.Result。 */
export type EvaluateContestAccessResult = {
  contestId: ContestId
  contestSlug: ContestSlug
  contestTitle: ContestTitle
  contestStarted: boolean
  contestEnded: boolean
  isRegistered: boolean
  containsProblem: boolean
  canViewContest: boolean
  canViewContestDetail: boolean
  canManageContest: boolean
  canViewLinkedContestProblem: boolean
  canManageLinkedContestProblem: boolean
  canSubmitContestProblem: boolean
}
